src/app/Validation/ValidationProblemDetailsResult.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Helium;
+using CSE.Helium.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Extensions;
+
+namespace CSE.Helium.Validation
+{
+    public class ValidationProblemDetailsResult : IActionResult
+    {
+        private readonly ILogger logger;
+
+        public ValidationProblemDetailsResult()
+        {
+            // todo: can this be done a different way so we don't need the ServiceActivator to get an instance of ILogger?
+            using var serviceScope = ServiceActivator.GetScope();
```

Comment:
Probably the easiest thing to do is put a public static read only property on App and have it call host.Services.GetService

Code here would look like:
logger = App.GetValidationLogger();

in app

private validationLogger;

public ILogger GetValidationLogger
if (validationLogger == null)
{
  validationLogger = host.Services.GetService ...
}
return validationLogger;

Reply:
When you say "App" do you mean in `Program.cs` (the sealed partial `App` class)?

Reply:
I'm going to leave the todo for now and look to refactor. The current PR satisfies the requirement and this may change if we implement the `LoggerMessage` delegates.